File: creational/python/builder/MediumPackCarBuilder.py
from CarBuilder import CarBuilder

class MediumPackCarbuilder(CarBuilder):
    def __init__(self):
        CarBuilder.__init__(self)

    def buildEngine(self) ->None:
        self._car.setEngine("w12")

    def buildInterior(self) ->None:
        self._car.setInterior("medium pack interior")

    def buildHeadLight(self) ->None:
        self._car.setHeadlight("lens")

    def buildSunroof(self) -> None:
        self._car.setSunroof("sunroof")

File: creational/java/builder/src/MediumPackCarBuilder.java
public class MediumPackCarBuilder extends CarBuilder{
    @Override
    public void buildEngine() {
        this.car.setEngine("w6");
    }

    @Override
    public void buildHeadLight() {
        this.car.setHeadlight("lens");
    }

    @Override
    public void buildInterior() {
        this.car.setInterior("medium pack");
    }

    @Override
    public void buildSunroof() {
        this.car.setSunroof("sunroof");
    }
}
